### PR TITLE
Fixed WebDefinition.IndexedPropertyKeys

### PR DESCRIPTION
**Pull request by andreasblueher**
Found a bug where given settings would be overwritten if something else had been already configured. Not sure why this can't be automatically merged though.

#749 #776

### CODE REVIEW

File: SPMeta2/SPMeta2.CSOM/ModelHandlers/WebModelHandler.cs [hunk 1]
```diff
@@ -397,9 +398,22 @@ private static void MapProperties(Web web, WebDefinition webModel)
...
+                // If other property keys have been set before, overwriting them would be bad
+                if (props["vti_indexedpropertykeys"] == null || props["vti_indexedpropertykeys"].ToString() == string.Empty)
+                {
+                    props["vti_indexedpropertykeys"] = GetEncodedValueForSearchIndexProperty(webModel.IndexedPropertyKeys);
+                }
+                else
+                {
+                    var indexPropertyValue = props["vti_indexedpropertykeys"].ToString();
+
+                    var indexList = GetDecodeValueForSearchIndexProperty(indexPropertyValue);
```
Comment (SubPointSupport) [inline SPMeta2/SPMeta2.CSOM/ModelHandlers/WebModelHandler.cs]: These won't work well.

 indexList.AddRange(webModel.IndexedPropertyKeys) would add same-same props over with the several deployments. 

indexList.AddRange(webModel.IndexedPropertyKeys); should include case-insensitive Contains() check if the props is there or not.

props["vti_indexedpropertykeys"].ToString() == string.Empty ->
string.IsNUllOrEmpty9props["vti_indexedpropertykeys"].ToString()), there is also w ConvertUtils class that can help.

vti_indexedpropertykeys needs to be shared in some kind of internal M2 class, there was a class with some OOTB property names, we'll look up it.

**Comment by SubPointSupport** [comment]
@andreasblueher FYI, added some notes into the pull request, not sure if there was a notification on that.

File: SPMeta2/SPMeta2.CSOM/ModelHandlers/WebModelHandler.cs (hunk 1 above)
Comment (andreasblueher) [inline SPMeta2/SPMeta2.CSOM/ModelHandlers/WebModelHandler.cs]: You're absolutely right. Looking forward to those ConvertUtils and if I remember correctly reSP suggested something similar for checking for null and string.Empty.

Still a lot of things I got to learn and I'm improving with every pull request ;)

Will you make the changes or can I add an other change to this pull request?

**Comment by SubPointSupport** [comment]
Please keep the branch in synch with the latest dev branch, so no merging will be required.
Always get the latest before developing new features, and let us know which features you develop.

No further broken pull request will be accepted.

**Pull request merged**